cmd/podman/buildfarm/remove.go

```diff
@@ -0,0 +1,94 @@
+package buildfarm
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/containers/common/pkg/config"
+	"github.com/containers/podman/v4/cmd/podman/common"
+	"github.com/containers/podman/v4/cmd/podman/registry"
+	"github.com/spf13/cobra"
+)
+
+var (
+	buildfarmRmDescription = `Remove one or more existing farms.`
+	rmCommand              = &cobra.Command{
+		Use:               "rm [options] FARM [FARM...]",
+		Aliases:           []string{"remove"},
+		Short:             "Remove one or more farms",
+		Long:              buildfarmRmDescription,
+		RunE:              rm,
+		ValidArgsFunction: common.AutoCompleteFarms,
+		Example: `podman buildfarm rm myfarm1 myfarm2
+  podman buildfarm rm --all`,
+	}
+
+	// Temporary struct to hold cli values.
+	rmOpts = struct {
+		All bool
+	}{}
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: rmCommand,
+		Parent:  buildfarmCmd,
+	})
+	flags := rmCommand.Flags()
+	flags.BoolVarP(&rmOpts.All, "all", "a", false, "Remove all farms")
+}
+
+func rm(cmd *cobra.Command, args []string) error {
+	cfg, err := config.ReadCustomConfig()
+	if err != nil {
+		return err
+	}
+
+	if rmOpts.All {
+		if cfg.Engine.Farms != nil {
+			for k := range cfg.Engine.Farms {
+				delete(cfg.Engine.Farms, k)
+				fmt.Printf("Farm %q deleted\n", k)
```

Comment:
I think that's too early as the `Write` may fail.

Reply:
Fixed